src/test.ts

```diff
@@ -1,20 +1,73 @@
-import { composeReducers, EnhancedReducerResult } from "./redux-reducer-effects";
+import enhance, { composeReducers, EnhancedReducerResult, getState, getTasks } from "./redux-reducer-effects";
 import { assert } from "chai";
+import { createStore } from "redux";
+import { Subject, Observable, Scheduler } from "@reactivex/rxjs";
 
-type Action = { type : string};
-type Task = { task: string };
+const create = <T>(t: T): T => t;
+
+type IncrementAction = { type: 'increment' };
+type AsyncIncrementAction = { type: 'asyncInc' };
+type Action = IncrementAction | AsyncIncrementAction;
+type AsyncIncrementTask = { task: 'asyncInc' };
+type StubTask = { task: 'do thing' };
+type Task = AsyncIncrementTask | StubTask;
 type State = {
     counter: number,
 };
 
 describe("redux-reducer-effects", function() {
 
+    describe("enhance store", function() {
+
+        it("can create", function() {
+
+
+            const enhancerStack = enhance({
+                createSubject: () => new Subject<Task>(),
+                taskRunner,
+            });
+
+            const enhancedCreateStore = enhancerStack(createStore);
+            const store = enhancedCreateStore(reducer, { counter: 0 }, enhancerStack);
```

Comment:
We should always construct our store via the enhancer instead of via the original `createStore` function, otherwise we lose some type safety. Consumers who don't care about type safety may use Redux's `createStore(enhancers)` method, but we do :-)